src/components/FeatureViews/AdminPortal/PanelManager.js

```diff
@@ -0,0 +1,306 @@
+            <Grid container alignItems={"center"}>
+                <Grid item xs={3} md={3} >
+                    <TextField
+                        value={record.name}
+                        defaultValue={record.name}
+                        label={"Name"}
+                        onChange={handleEditRecord("name", record.id)}
+                    />
+                </Grid>
+                <Grid item xs={7} md={7}>
+                    <TextField
+                        value={record.description}
+                        defaultValue={record.description}
+                        label={"Description"}
+                        onChange={handleEditRecord("description", record.id)}
+                    />
+                </Grid>
+                <Grid item xs={2} md={2}>
+                    <Button
+                        onClick={editRecord(record.id)}
+                        className={"button"}>
+                            UPDATE
+                        </Button>
+                </Grid>
+            </Grid>
+        </Paper>
+    )
+}
+    const generateGrid = () => {
+        /**
+         * Generates header with fields
+         */
+        
+        const headerElements = [];
+        for (const [index, value] of fieldsWithDefaults.entries()) {
+
+            headerElements.push(
+                <Grid item xs={value["col-width"]} md={value["col-width"]} >
+                    <Typography align={value["align"]} style={{color: 'white', fontWeight: '500'}}>
+                        {value["name"]}
+                    </Typography>
+                </Grid>
+            )
+        }
+
+        return (
+            <Grid container>
+                {/* header */}
+                <Grid item xs={12}>
+                    
+                    <Grid container className={'accounts-table-heading'}>
+                    {headerElements}
+                    <Grid item xs={2} md={2}>
+                        <Typography align={'center'} style={{color: 'white', fontWeight: '500'}}>
+                            Edit
+                        </Typography>
+                    </Grid>
+                    </Grid>
+                </Grid>
+
+                {/* Directory */}
+                <Grid>
+                    <Grid item xs={12}>
+                        <Grid container spacing={8} alignItems={"center"}>
+                            {
+                                recordList.length > 0 ? recordList.map((record) => {
+                                    console.log(record);
+                                    return (<Grid item xs={12} md={12} key={record.id}>
+                                        {
+                                            record.editing ? editRecordRow(record) : viewRecordRow(record)
+                                        }
+                                    </Grid>);
+                                }): <NoListAlert list={"Course Categories"} />
```

Comment:
`<NoListAlert/>` will display an alert stating the `list` doesn't have any item. Right now all it will display is "There are no Course Categories" even if it's a discount or a tuition rule since it's hard coded in here.  This should be a prop that's passed in.